src/stdatamodels/validate.py

```diff
@@ -35,6 +39,100 @@ def value_change(path, value, schema, ctx):
     return update
 
 
+def _validate_datatype(validator, schema_datatype, instance, schema):
+    """
+    This extends the ASDF datatype validator to support ndim
+    and max_ndim within individual fields of structured arrays,
+    and handle the absence of the shape field correctly.
+
+    Additionally, dtypes are required to be equivalent, instead
+    of just "safe" to cast.
+    """
```

Comment:
Is there a reason to have this extend the validator just in `stdatamodels` and not have it available in `asdf`?  Or do you see this as something that might be moved over to `asdf` at some point?

Reply:
I think we'll want to shore up the asdf validator eventually, but I'm not prepared to do that now -- it's going to cause havoc since previous valid files will become invalid.